fashion_mnist/README.md

```diff
@@ -0,0 +1,394 @@
+
+```python
+# Horovod: print logs on the first worker.
+verbose = 1 if hvd.rank() == 0 else 0
+```
+
+![image](https://user-images.githubusercontent.com/16640218/53534314-2244c500-3ab3-11e9-95ef-e7e7b282ab4f.png)
+
+### Read checkpoint only on first worker
+
+For the same reason as above, we read checkpoint only the first worker and *broadcast* initial state to other workers.
+
+Replace the following code:
+
+```python
+# Restore from a previous checkpoint, if initial_epoch is specified.
+if resume_from_epoch > 0:
+    model = keras.models.load_model(args.checkpoint_format.format(epoch=resume_from_epoch))
+else:
+    ...
+```
+
+with:
+
+```python
+# Restore from a previous checkpoint, if initial_epoch is specified.
+# Horovod: restore on the first worker which will broadcast both model and optimizer weights
+# to other workers.
+if resume_from_epoch > 0 and hvd.rank() == 0:
+    model = hvd.load_model(args.checkpoint_format.format(epoch=resume_from_epoch))
+else:
+    ...
+```
+
+![image](https://user-images.githubusercontent.com/16640218/53534410-9717ff00-3ab3-11e9-86eb-1bf8299416d2.png)
+
+### Adjust learning rate and add Distributed Optimizer
+
+Horovod uses an operation that averages gradients across workers.  Gradient averaging typically requires corresponding increase in learning rate to make bigger steps in the direction of a higher-quality gradient.
+
+Replace `opt = keras.optimizers.SGD(lr=args.base_lr, momentum=args.momentum)` with:
+
+```python
+# Horovod: adjust learning rate based on number of GPUs.
+opt = keras.optimizers.SGD(lr=args.base_lr * hvd.size(),
+                           momentum=args.momentum)
+
+# Horovod: add Horovod Distributed Optimizer.
+opt = hvd.DistributedOptimizer(opt)
+```
+
+![image](https://user-images.githubusercontent.com/16640218/53534579-52409800-3ab4-11e9-971e-f7def73c7b36.png)
+
+### Add BroadcastGlobalVariablesCallback
+
+In previous section, we mentioned that first worker would broadcast parameters to the rest of the workers.  We will use `horovod.keras.BroadcastGlobalVariablesCallback` to make this happen.
+
+Add `BroadcastGlobalVariablesCallback` as the first element of `callbacks` list:
+
+```python
+callbacks = [
+    # Horovod: broadcast initial variable states from rank 0 to all other processes.
+    # This is necessary to ensure consistent initialization of all workers when
+    # training is started with random weights or restored from a checkpoint.
+    hvd.callbacks.BroadcastGlobalVariablesCallback(0),
+
+    ...
+```
+
+![image](https://user-images.githubusercontent.com/16640218/53535043-1dcddb80-3ab6-11e9-9911-1eb33a1f531c.png)
+
+### Add learning rate warmup
+
+Many models are sensitive to using large learning rate immediately after initialization and can benefit from learning rate warmup.  The idea is to start training with lower LR and gradually raise it to target LR over few epochs.  Horovod has convenient `LearningRateWarmupCallback` for Keras API that implements that logic.
```

Comment:
large learning rate --> a large learning rate (LR)
raise it to --> raise it to a 
over few --> over a few
Horovod has --> Horovod has the 
for Keras API --> for the Keras API